### ISSUE TITLE
[Discussion] OpenToonz development

### ISSUE THREAD

**Issue by shun-iwasawa** [opentoonz/opentoonz#2746]
Dear OpenToonz community members,

I post this comment is to let you know my recent situation and to clarify a policy for reviewers to judge which PR can be merged to OT.

I had consulted on management of open-source community of OpenToonz as "paid work" since it had been released. The situation has changed a month ago, I started a new career and now I'm working on a voluntary basis.

Thanks to many motivated and highly-skilled contributors, many wonderful PRs are already waiting for review, but I worry about that it may need more time to be merged in the future because I can work on them during night-time and weekend.

It is not my intention to interrupt the progress of the software due to my delay. Therefore, here I would like to suggest a basic policy for development of OpenToonz so that other members can review and merge more PRs without hesitating.

The policy consists of one principle and two supplementary notes:

- PR can be merged if it will not interfere existing usages.

note, however, that

1. If there is the possibility that introducing PR may conflict OT's existing license, please wait for owner's review.
1. From several month before releasing stable release date (to be announced), please do not merge "feature PR"s.

OT has many features which enables various way of producing animation.
A modification for some workflow may become disadvantage for others. In such case please consider to make it optional (via user profiles) so that users can select whether to use it.

Here are examples:

- Bugfix : Fine to be merged.
- Refactoring : Fine to be merged.
- Enhancement / Speed-up : Fine to be merged.
- Translation : Fine to be merged.
- New Fx : Fine to be merged.
- New command : Fine to be merged.
- New panel : Fine to be merged.
- New preset (fx, camera settings, shortcut, etc.) : Fine to be merged.
- Altering default user profiles (preferences, panel layout and shortcut) : Fine to be merged.

- Altering UI which cannot be controlled user profiles (like adding buttons to the panel or adding commands to the context menu) : Basically it's fine to be merged, but please be careful if it will not disturb the conventional usage.
- Altering existing fx or rendering behavior : Basically it's fine to be merged, but please ensure that it will not change a rendering result of a scene which is made with the previous version.

As always, any contributions are much appreciated.
Please let me know if there are any opinions or questions to the above.

Thanks!

**Comment by turtletooth** [comment]
@shun-iwasawa Is there a way to assign Jenkins privileges to other developers?

**Comment by gab3d** [comment]
@shun-iwasawa 

> Please let me know if there are any opinions or questions to the above.

a million questions! 😄 
first off, thanks for your openness and about the actual status of OT development, and for all your contributions to the project (past, present and hopefully future ones).

I think OpenToonz is an open source project that is really worth it, because it has a very solid foundation, and although needing a lot of work to modernize some of its infrastructure, it gives the community an excellent starting point from where to grow. 
But what I also see is that for that to really happen, the project needs some way of organize itself to be able to set goals and dedicate resources to achieve them (and for that, there has to be resources in first place). But I think it all starts with the planning and the organization...

So, one of the things I'd like to know is if there's currently somebody/organization leading that way, and if the answer is no, well... perhaps we have to manage ourselves to create some kind of *OpenToonz Foundation* (to say something) to be the place where to centralize the efforts behind the project development, so we can reach out the community and ask for support for specific project goals, to be able to attract both donations and programmers. 
Perhaps applying for Google Summer of Code or something along that lines would also be a great benefit.

**Comment by shun-iwasawa** [comment]
@turtletooth
I'll try modifying Jenkins settings so that it will run on posting the PR without the chant "Jenkins" and see if how it will be a burden. 

@gab3d 
> all your contributions to the project (past, present and hopefully future ones).

I'm sorry I didn't explain it enough, but I will continue to *develop* OpenToonz as ever. :-)
Let me clarify about my new career - on the end of July I joined Studio Ghibli (again) to produce Miyazaki-san's new film. Since they are using OpenToonz for their production I'll develop OT according to their requests during working hours.

On the other hand, I'll also continue to manage OT open-source community as ever.
Dwango and I made an (unpaid) agreement so that I can continue the management like releasing the future stable versions and updating the official website on behalf of them.
They will be basically executed at home after working hours.

I wrote the post because I see that the urgent issue of the project would be a pile of pending PRs.

**Comment by manongjohn** [comment]
@shun-iwasawa 
Just a suggestion...In order to ensure that PRs that might impact the studios don't get merged without further scrutiny and/or you would like to personally review it, could mark it in such a way that we know not to merge without your consent? Perhaps tag yourself as a Reviewer so we know?

You don't necessarily have to review it right away.
Other can still review it to help out, but you would have the final say in it.

Conversely, when devs submit PRs, they should add you as a reviewer if they feel it requires your attention. You can always remove it if not needed.

**Comment by gab3d** [comment]
@shun-iwasawa 
I'm so glad to know you'll still be very involved in OT's development; and for sure, what you happen to develop for Ghibli will also be very useful for the rest of us.  :))
So, that's great!

Going to the organization of the community and extra development efforts (besides what you somewhat "officially" do), do you feel there's room for building up some kind of structure, where Dwango and other interested people and supporting organizations could participate to develop a future path for the application? 🤔

**Comment by shun-iwasawa** [comment]
@manongjohn
Thank you for the suggestion, I'll do so when it's necessary. 
Please keep in mind that the reverse is not always true. Being without the reviewer tag does NOT mean that it is accepted to be merged.

@gab3d
I would like to be always open to any suggestions, but sorry I'm still not quite sure what you actually mean the "organization" is.
You wrote : "the project needs some way of organize itself to be able to set goals and dedicate resources to achieve them". Well. I'm not sure (1) why do you think it is needed, (2) what do you think the goal would be, and (3) how do you think the organization can make community to agree with, and dedicate for specific goals?
Currently the members contribute the software in many way with their own motivation and purposes. Some of them (such as Morevna project) are sharing the purpose in group. The software can grow in any way like amoeba. I think it is one of the advantages of the open-source development. I'm concerning about that pointing in a certain path may make some of them to get demotivated.

**Comment by shun-iwasawa** [comment]
@gab3d 
Sorry I think I was a little stubborn in the previous comment.
I agree that there are successful open source projects with organization and it's important to think of any possibility to achieve a better community in the future.

However, as you pointed, for building up some kind of structure we need careful discussions with Dwango, with all contributors, and in the community. I think especially the discussion with Dwango will be my responsibility. It must be a tough work and currently - I'm sorry to say - I don't have time to tackle with it.
I think I'll have the elbowroom as my sons grow up...

**Comment by gab3d** [comment]
@shun-iwasawa 
I really appreciate both your messages.
I think I understand that both Dwango and you personally are, and should be, in any scenario, pivotal blocks of the community and development behind OpenToonz.

That said, I suspect that the human resources dedicated to manage and plan goals for the application for the future, and also those dedicated to reach out people (i.e. communication in general, but also funds raising) are sparse, and probably most of the people behind it (i.e. you 😉 ) also have to share a good piece of their time, dedicating efforts to specific coding tasks.

So, my main idea here is asking if we could somehow help to inject some human life into several of these tasks, to make them go ahead at a greater pace or sometimes more consistently... or who knows...! But, I'm sure something positive could be done in a lot of fronts.

With respect to Dwango and the current organization behind the OT project, we can't know much from the outside about their own goals with it, or their way to approach it as a company, but nevertheless it's not the intention here that they have to resign any of their main goals, but instead for more people to be able to help make that goals (and other complementary ones) possible at a faster pace, by bringing in the equation some more resources to make them all happen.

I know these are not easy questions to answer for you alone, nor is that I expect any immediate actions of any type, but just let you (and Dwango) know there could be help aboard to make OpenToonz reach the best destiny possible. 
Just ring the bell and let us know you are interested!  👍 

Thanks for listening!

**Comment by RodneyBaker** [comment]
One question I am curious about is whether any specific targets might be identified going forward into the v1.4 release, perhaps especially as we are approaching that release timeframe.  The obvious target will be to collect all the nightly releases into that new official release.  Above and beyond that however it's not clear what the needs of the community are beyond that.  The currently stated foundation for Opentoonz (from the main website) is:

> - Software for the production of 2D animation.
> 
> - Based on the software "Toonz", developed by Digital Video S.p.A. in Italy, OpenToonz has been customized by Studio Ghibli, and used for the creation of its works for many years. Dwango has launched the OpenToonz project in cooperation with Digital Video and Studio Ghibli.
> 
> - OpenToonz can be used free of charge for both commercial and non-commercial projects.
> 
> - Available as an open-source so that anyone can modify its source code freely.
> 
> - We aim to develop a new platform for connecting the academic research with cutting edge animation production.

We can all understand the first four bullet items and hopefully appreciate them.  It is likely the last one where most of us as users/aspiring developers start to fade as this area falls more into the realm of efforts outside our capability.  We don't know what academic research is much less know how to define what cutting edge animation production might be.  So this is an area that Dwango, Digital Video and Studio Ghibli define.  There currently isn't a bullet point for goals beyond those authorities.

A starting point is suggested however in that we can research and specify what production solutions we need more broadly and determine how well they fit into the current Opentoonz development cycle.  

This discussion is appreciated as it gives a solid framework for moving forward to the v1.4 release.

**Comment by jrhyme** [comment]
I don’t want to seem rude, it’s painful, but I’ll say what I think .. the goal of the program is to become the coolest? or just to please japanese studios? I really like this software and I don’t want to leave, I am also grateful to all the guys who voluntarily contributed to the development, and to you Shun, for spending free time, but if I were a developer, I would give up developing because there are no ideas accepted. What is the further development path? What awaits the official version, I read the posts above, but everything is vague. I apologize again.

**Comment by manongjohn** [comment]
@shun-iwasawa, et al...

Perhaps this is a good time to discuss having a "Community Edition" of OpenToonz (OTC)?

The Community Edition would be managed and driven by the community without concern of impacting existing studios currently using OpenToonz.  The original version would maintain it's studio focus yet still potentially benefit from changes made in the Community edition.

We would need to find someone to manage the Community version (no, I am not nominating myself!) while you continue to manage the original version.  You wouldn't have to worry about what gets merged in the Community version and can focus only on changes you need for the original version and any changes you want to port over from the Community.

I would like to see it still fall under the OpenToonz organization in it's own repo (i.e. github.com/opentoonz/opentoonz_community) and adhere to any licensing constraints and operating parameters Dwango puts forth.

The benefit of a Community edition is that we would no longer be constrained by potential impacts to current studios.  We can change things as we feel needs to be done to improve OT from multiple aspects, not just a studio viewpoint. 

Changes can be shared between both projects as needed. You would control the official version and incorporate only changes you want from the Community edition as you see fit. The Community edition would incorporate official changes as they become available. The good thing about that is if a studio had been reluctant about certain changes but sees its usefulness is in the Community edition, it can be requested to be ported in.

The drawback would be the potential for the code base to diverge so much so that it becomes almost impossible to share code.

A Community edition may also attract more developers if they know they are free to develop without being restricted.

I'm sure there are other benefits and drawbacks to discuss if we decide to go this route. 

Technically this can be done outside of Dwango, but I think they can benefit from this as well.

Thoughts?

**Comment by jrhyme** [comment]
> @shun-iwasawa, et al...
> 
> Perhaps this is a good time to discuss having a "Community Edition" of OpenToonz (OTC)?
> 
> The Community Edition would be managed and driven by the community without concern of impacting existing studios currently using OpenToonz. The original version would maintain it's studio focus yet still potentially benefit from changes made in the Community edition.
> 
> We would need to find someone to manage the Community version (no, I am not nominating myself!) while you continue to manage the original version. You wouldn't have to worry about what gets merged in the Community version and can focus only on changes you need for the original version and any changes you want to port over from the Community.
> 
> I would like to see it still fall under the OpenToonz organization in it's own repo (i.e. github.com/opentoonz/opentoonz_community) and adhere to any licensing constraints and operating parameters Dwango puts forth.
> 
> The benefit of a Community edition is that we would no longer be constrained by potential impacts to current studios. We can change things as we feel needs to be done to improve OT from multiple aspects, not just a studio viewpoint.
> 
> Changes can be shared between both projects as needed. You would control the official version and incorporate only changes you want from the Community edition as you see fit. The Community edition would incorporate official changes as they become available. The good thing about that is if a studio had been reluctant about certain changes but sees its usefulness is in the Community edition, it can be requested to be ported in.
> 
> The drawback would be the potential for the code base to diverge so much so that it becomes almost impossible to share code.
> 
> A Community edition may also attract more developers if they know they are free to develop without being restricted.
> 
> I'm sure there are other benefits and drawbacks to discuss if we decide to go this route.
> 
> Technically this can be done outside of Dwango, but I think they can benefit from this as well.
> 
> Thoughts?

I absolutely agree, we do not want the software to be highly specialized, you need to make it top-notch)

**Comment by turtletooth** [comment]
I like the idea of a community version very much, but I think the reality would eventually become very difficult to manage in parallel with the main version.  As @manongjohn mentioned eventually the code bases would/could diverge so much that bringing features back and forth would become difficult.  

For example, the assistants tool was built off an earlier version, and resolving the conflicts to get it up and running with the current code was a pretty big feat.

To keep developers around and inspire new involvement, I do think the pace of reviewing and merging needs to be greatly increased.  

If a community version is developed, it may benefit from not being under Dwango's umbrella.   For example if we use certain libraries that are compatible with the MIT license, we would have the freedom to use them without waiting to hear from the Dwango team to approve or reject the idea.

There are plenty of successful open source projects that thrive under the umbrella of a parent company such as Visual Studio Code or Chromium, but usually those projects are receiving much more attention from the company.  It sounds like @shun-iwasawa is the sole liason between Dwango and the rest of the community and he now is both quite time constrained and no longer an official part of Dwango.  So OpenToonz is left with an affiliation to a company that isn't (please correct me if I'm wrong) taking an active part in the software itself.

There is certainly discussion to be had around this.

**Comment by gab3d** [comment]
> It sounds like @shun-iwasawa is the sole liason between Dwango and the rest of the community and he now is both quite time constrained and no longer an official part of Dwango. So OpenToonz is left with an affiliation to a company that isn't (please correct me if I'm wrong) taking an active part in the software itself.

That's also my impression (but that could be wrong, of course). The fact is that aside from Shun's posts, we have no other news about Dwango or their ideas for the software development.
In fact, the root of my request for discussion (in the way I saw it, at least), was more as to empower the development, planning and communication of the project, for achieving both, Dwango/japanesse studios needs and other interested parties (not affiliated with them) goals too.

With respect to forking:
Personally I feel we could gain a lot more in maintaining a single version of the software, by just improving the revision/approval process (Shun's been quite open about it in his last post), and by being able to clarify the main "next" goals of development for the application, so we (OpenToonz) can start to devise the best ways to raise the needed resources for making them happen in a reasonable time-frame.

For example:
I don't think Dwango, Ghibli, etc. could complain about things like:
- a modernization of the viewport so we could have:
  - realtime feedback for things like blending modes, masking and maybe certain effects.
  - an overall speed improvement by using the full power of modern GPUs.
- improving complex animation capabilities, mainly:
  - being able to refer to objects into the main xsheet and sub-xsheets by name.
  - being able to have expressions act as a modifier "on top" of standard (keyframed) animation and not replacing it altogether, as now.
  - being able to have pegbars at any level of a hierarchy, to use them as null objects in complex rigs.
  - being able to create an effects with custom sliders, values, colors, for using in rigging.
- perhaps even being able to accept OpenFX as another plug-in for effects, as it would open the doors for more existing effects.
- or improving the native Particles effect, which is an interesting base also, but needs more features.
- more GPU effects.
- improving the point tracker (Blender has a excellent one that perhaps could be used as reference).
- improving the SVG Import.
- having a proper Pen Tool.
- etc.

This is not intended to be anywhere near a complete list of features, but instead as a sample of the kind of things that could benefit everyone across the board, without breaking anything that makes OT so useful for Ghibli or other current users.
And still they are tasks that :
 - are quite important for the future of the application.
 - will take much more manpower than we currently have available.

So is for that reason that I think if we could agree on some of these kind of things as truly needed features, then we could impose a priority and make efforts pulling them out with much more impulse than we currently have. 🤔 

Just trying to clarify what moved me to bring this issue to the table...

**Comment by jpturcotte** [comment]
Hello everyone,
Long time no see (which is entirely my fault)! I just wanted to give my two cents on the very few areas I believe my opinion might be informed and of enough value. First, congratulations to Iwasawa-san for his new job, and thanks to everybody who actually stepped up to help maintain this great piece of software. Second, regarding these particular points:
- Community Edition: as seducing as it sounds, I agree it would amount to a soft fork. From what I know of you great people, this CE would probably become the preferred version overtime. Thus, it can and should be done only if Dwango cannot provide the required leadership, and it would mean taking on the future of OT as a community.
- OpenFX integration: it is not happening. The OpenFX forum called for help to push a new revision out the door at the beginning of the year, without much success. The people who coded the solutions seem to have moved on. Natron is on life support. There is not much hope other than someone at a big company or university raising their hand to steward the standard. It is sad, because the call for help estimated the work to publish an update to about 10 hours a month.
- Better SVG support: before my disappearing act, I was researching this. I know who to ask, and I could get an estimate if the community considers this a priority. Basically, Qt's support is borked. This would be huge.

**Comment by shun-iwasawa** [comment]
@manongjohn

If the reason of your suggestion to have another edition was to develop the software more freely, I think you can do it here in the original version.

I wrote in the first post as `PR can be merged if it will not interfere existing usages`. It means that, when you introduce some feature, the software should also leave the option to act as the same as ever, if there may be users who will not need the feature in their workflow.
To be more specific, when you would like to introduce a feature for specific usage or for part of users, it is desirable to be able to deactivate / hide via user profiles.
(You know, it's just the same strategy what we have done up to now!)
And you can set the option to active by default if you think it would be helpful for new users.

The policy is not intended to prioritize Japanese animation industry, but to respect a diverse usages done by users around the world.

Of course everyone is free to fork the repository based on the license. But I personally think that the software will be better by developping the original version together, rather than by separating the edition.

**Comment by Lovewyrm** [comment]
>community edition

Ah, nightmares of Cinelerra come back. 
Please don't. I'd rather be patient on Dwango OT stuff than splitting this up even more.
That's my own opinion, of course.

**Comment by jrhyme** [comment]
> > общественное издание
> 
> Ах, кошмары Cinelerra возвращаются.
> Пожалуйста, не надо. Я предпочел бы быть терпеливым к вещам Dwango OT, чем разделять это еще больше.
> Это мое собственное мнение, конечно.
> 
> PS:
> Toonz начал гораздо более ограниченно и все такое, и мы все равно получили это прекрасное программное обеспечение, как и сейчас.
> Конечно, открытие - это новый стандарт с новыми ожиданиями разработчиков и пользователей, но, на мой взгляд, все же стоит помнить.
> 
> ... Мне просто не нравятся версии сообщества.

We are not conservatives, and we are not happy with the situation, because we want to develop and popularize software and attract developers, I was a little angry with your message.

**Comment by Lovewyrm** [comment]
I just don't like "Project" Community Edition.
Half of my worries would be gone if the project got an entire different name, simply because documentation, web searches, etc SUCK the moment something is duplicated in name. Fedora is Red Hats community edition, right? 
Imagine if Fedora was called Red Hat Community Edition? You'd have an extra layer of dirt to dig through because Red Hat "Not Community Edition" would interfere with web searches.

And Fedora has a pretty beefy team behind it, more or less.
Is this the case in the community vs Dwango? I doubt it.
I'd really rather see Dwango be persuaded to pull in some mediator, even if they'd prefer Ghiblis or Dwangos decisions, it would give the community a hook to either pursue or not pursue.

If they pursue the suggestions from Dwango, then OT would get better, too, and mediator implies a bit of a two way street, too.
OT is already pretty good and the improvements should tell any sane person over at Ghibli and co that the community isn't just a leech.

I'm sure there can be some sort of bridge. I don't like tenebrious overlords, myself, but in the case of OT, they even made it open to begin with, so the folks behind it are not entirely evil.
It's not GPL or anything, but I'd rather see some negotiations first than jumping into a completely different version, especially if it shared the same name.
If you call it something different, then fine, that way if the community version gets that surge of activity and perhaps loses it, then at least the "OpenToonz" version can still be searched for, which is already a good and working suite, without things interfering.

But yeah, this is just an opinion, I have dabbled with OT code myself, so I could technically become a contributor but overall it's best to regard me as useless dirt on that front. I'm a user first and foremost and I've been raging at duplicated confusing crap before. I ...really don't like it. I hate wading through crap.

**Comment by jrhyme** [comment]
> I just don't like "Project" Community Edition.
> Half of my worries would be gone if the project got an entire different name, simply because documentation, web searches, etc SUCK the moment something is duplicated in name. Fedora is Red Hats community edition, right?
> Imagine if Fedora was called Red Hat Community Edition? You'd have an extra layer of dirt to dig through because Red Hat "Not Community Edition" would interfere with web searches.
> 
> And Fedora has a pretty beefy team behind it, more or less.
> Is this the case in the community vs Dwango? I doubt it.
> I'd really rather see Dwango be persuaded to pull in some mediator, even if they'd prefer Ghiblis or Dwangos decisions, it would give the community a hook to either pursue or not pursue.
> 
> If they pursue the suggestions from Dwango, then OT would get better, too, and mediator implies a bit of a two way street, too.
> OT is already pretty good and the improvements should tell any sane person over at Ghibli and co that the community isn't just a leech.
> 
> I'm sure there can be some sort of bridge. I don't like tenebrious overlords, myself, but in the case of OT, they even made it open to begin with, so the folks behind it are not entirely evil.
> It's not GPL or anything, but I'd rather see some negotiations first than jumping into a completely different version, especially if it shared the same name.
> If you call it something different, then fine, that way if the community version gets that surge of activity and perhaps loses it, then at least the "OpenToonz" version can still be searched for, which is already a good and working suite, without things interfering.
> 
> But yeah, this is just an opinion, I have dabbled with OT code myself, so I could technically become a contributor but overall it's best to regard me as useless dirt on that front. I'm a user first and foremost and I've been raging at duplicated confusing crap before. I ...really don't like it. I hate wading through crap.

How naive. No comments.

**Comment by turtletooth** [comment]
I'm going to close this.  I think @shun-iwasawa has addressed the concerns raised and has presented a very acceptable plan for both PRs and the future of the software.  I agree that we are stronger together than apart.  

I think the big frustration for some developers was the pace of merging PRs.  I think we have a plan now that should address that.

**Comment by manongjohn** [comment]
I wouldn't necessarily say it was just the pace of merging PRs. 

With all due respect, OpenToonz caters to  the Japanese studios that use it.  I get it...it's where it started, so to speak.  As such, changes are not always entirely welcomed right away. I know I had at least 1 PR that needed to be checked to see if that change would be accepted.  I think that was because they originally requested to remove it and I was bringing it back because users were requesting it and it made sense.  But think about that... a positive change for the community was potentially being thrown away because the studio was there first and has more say in what goes into OT than the general community. Luckily the powers that be agreed in this case.

But I'm not the only one who feels this way from small conversations here and there.

Keep in mind, how OT is integrated into the workflow of current studios may not fit with other studios.  If we have to be constantly aware of not making changes that would alter existing workflows, then we are by definition being restricted.  A community edition could make changes that could alter a workflow, but if it works to include other studios or is actually an improved way of doing it. Isn't that a good thing?

At the same rate, I understand creating yet another version can be a lot of work and provide a lot of confusion.  Tracking and maintaining OTX, alone, is no easy feat when keeping up with the few developers we do have. I can't imagine what it would be like if there were a lot more.

I think some of the issues can be addressed with good planning and communication between the 2 sides, though.  For example, if a new feature is being introduced and requires a change to core, that core change should be identified, separated and shared between both versions so the base is relatively the same.  This is a good thing if that core change makes something more proficient and more adaptable for other features.

Having said all that, it does not appear Dwango would support a Community version so the discussion ends here, though were they even approached with the idea? Probably not. If a Community edition is to be created at this point, it would be brand new, somewhere else and likely called something else.

So for now, we should press forward with OT as it is and see where the road takes us....

**Comment by jpturcotte** [comment]
Just to be clear: legally, Dwango and Digital Video can close back the source, but what is actually available under the BSD license will continue to be as long as the license and copyright notices are not changed. Anybody can fork it under a new branding without needing to ask for permission.

**Comment by beeheemooth** [comment]
I want to note that OpenToonz plays a large role not only in the field of animation production, but also in the field of education, since being a free program, it can be used in schools, institutes, sections, circles for teaching animation.
 This, in my opinion, is the main driving factor in the popularization of this software.
 Thus, OpenToonz is indirectly also a good example of advertising company representing it. Obviously, an increase in the number of users will lead to an increase in the rating of the company. Whether this rating will be positive or negative directly depends on what the response to users' requests will be, so I’m sure that any self-respecting company will take this important factor into account.
Thanks.

**Comment by jrhyme** [comment]
Predictably, it was possible not to open this topic.

**Comment by turtletooth** [comment]
@jrhyme Let's try and keep the conversation constructive.

**Comment by jrhyme** [comment]
> @jrhyme Let's try and keep the conversation constructive.

It was a cry of the soul and I deleted my messages, so as not to spoil the atmosphere. Why did the topic come up?

**Comment by manongjohn** [comment]
> Why did the topic come up?

That's probably my fault. Sorry. 😕

**Comment by gab3d** [comment]
I think forking is not the immediate answer to the current problems we may have now, plus it comes to a potentially unmanageable cost in fragmentation of an already small community.

As @shun-iwasawa has stated, there is now a good established policy about what can be safely merged into master, and I think this is a **lot** more, from any coder's point of view, than what we had previously.

On other fronts, I'd still like to be able to push a little on (us all) being able to produce:
- a public development plan for the application, at least a basic Roadmap.
- a plan for promoting OT
- a plan for funds raising, to be able to have certain things (from the Roadmap) be done by proficient programmers in certain technologies, that we currently don't have onboard.
...but I think for that to happen we should need at least a guidance from Shun, as he is the best positioned one for giving all that a somewhat "official" place to happen, inside the project structure, and not just be what a bunch of guys came up with by themselves.

**Comment by konero** [comment]
Actually Toonz was used all over the world, not just in Japan. We used it on Legend of Korra and Totally Spies, it was also used on Spirited Away, Futurama, Steven Universe, Arriety, Phantom Boy, When Marnie Was There and many more, what these works have in common is they’re traditionally animated and that is just what Toonz core user group happens to be.

This leaves me confused as to what the complaints have been here, the tools are there if you want to produce work and they are production proven, actually when OpenToonz first went public (after palettes were fixed) it was more developed than Harlequin was. The repository owner has also expressed they don’t really care too much so long as current usability and compatibility remains when it comes to contributing.

Now nobody has substantiated what “priorities, goals, developments” mean, what's missing? I hope it’s not a code word for moving OpenToonz away from traditional animation, that would be very bizarre to do.

If you want my advice, OpenToonz should play to its strengths, not every software can boast that some of the greatest works of art in cinema history were produced with it, go with it.

I work at Japanese companies yes, but also French and Korean as an animator, since the early millennium the painting process has not changed much at all, which is why in #690 I detailed something different you can read [here](https://github.com/opentoonz/opentoonz/issues/690#issuecomment-503597784) and [here](https://github.com/opentoonz/opentoonz/issues/690#issuecomment-504679020), but I need developers to build it as it’s beyond my ability, the viability of it remains unknown so it's a time waste risk. All this talk of SVG, rigging etc doesn’t come up in conversation, we can just choose something that already does what we want so getting investment in any form is difficult, what we’re on the look out for is solutions to problems that don’t exist yet. CACANi is still a buzz by the way, to use it as an example.

So perhaps you may want to allow feature requests again, this way student programmers looking for something to do are not only presented with a long list of bug reports for a software they are yet to dabble in, making reproducing issues much more painful. I know we got Google groups, but coders are unlikely going to visit it.

But don’t take this to heart too much, it’s just my perspective and experience, others will have differing ones no doubt.

**Comment by RodneyBaker** [comment]
Well said @konero .

You said:

> ...All this talk of SVG, rigging etc doesn’t come up in conversation...

SVG and rigging discussions come up pretty frequently in conversations in discord etc. where people would like to use Opentoonz along with other software but can't because of various incompatibilities.  

SVG
SVG is a really good example because it almost works!  Some of it even works really well.
Some of those issues are well known and the frustration real because those who would use SVG can see that little emphasis has been placed on fixing those issues.  Lest anyone think a lot of knowledge is in vector graphics is needed here to fix these I'd say with the exception of one... vector color fills, the major issues require none.  We could detail those in another report.  
Why develop SVG?  Because the Toonz PLI format is a black box (proprietary) format that cannot be used in other programs without conversion.  Note however that SVGs going out of Opentoonz work quite well.  As with other file formatting issues it is often importing that is the greater concern.

Rigging
Rigging is a longer development path as Opentoonz will need a lot of work to build the infrastructure needed and direct compatibility with external rigging systems is unlikely.  I'd say the best development approach here is likely going to be one of incremental change.  For instance, adding a framework for constraints in Opentoonz via Schematic nodes.  This would not just be useful to cutout animators but also useful for more traditional approaches.   This can be discussed elsewhere so I won't go into other details here except to say many constraints already exist say in the form of Pegs but users aren't aware of them.  Others such as "Aim at (Look At)" constraints would need to be developed.  This is also a good place where those that use other tools/applications could share how they use those to supplement work with Toonz/Opentoonz.  

This seemed a good opportunity to suggest how important these two areas (SVG and Rigging) are to the larger Opentoonz community.  Not so much to gain the full capabilities of other programs but for solid general usage compatible with both traditional and digital workflows.

**Comment by jrhyme** [comment]
In short, what have we come to? What would be the most ideal option? everything is vague.

**Comment by manongjohn** [comment]
Not to belabor this point much longer because it isn't going anywhere and, for the most part, people have responded to it not being a good idea.  Fair enough. It is just a discussion after all to see what people think.

Chalk this up to my naivete for even bringing it up. I know I'm not the only one who has concerns about working on OpenToonz, but I do anyway.  Let's leave it at that.

As for direction...we need direction and more developers willing to work on that direction.

Not sure what to do to attract more developers. Bounties didn't really do much and the 1 developer who tried to work on a small bounty didn't finish because what we wanted and what he was going to give us wasn't quite the same.  Perhaps we need clearer and better requirements?

Direction/Priority wise: We've done quite a bit to fix the stability and improved some UI/UX issue which is great.  I still see a number of users having issues or complaining about Vector levels, mostly with fills and occasional crashes on vector levels.  This really needs to be fixed or redone.  How can we be taken seriously if something as simple as a fill is problematic?

From the few issues I've looked into I think a lot of that might have to do with the detection and management of stroke and stroke-intersection data. The problem with isolating the root issue with this is the problems are hard to reliably reproduce making it difficult to see why.  In general I think that stroke intersection data should be reviewed and possibly redone.

I'm still think since we're having some issues because we aren't fully on OpenGL which needs to be implemented properly and tested thoroughly to avoid issues that happened with 1.2 release.  I believe we're waiting to see how Qt handles Apple's move to Metal? But how long must we wait?  I think we should just move to OpenGL now and then convert over later instead of trying to bypass it completely.

I'm sure there are many other pain points and improvements that need to be made but without clear leadership and a few more experienced developers, we're not going anywhere.

Anyway...just my naive thoughts, for what it's worth...

**Comment by jrhyme** [comment]
> Not to belabor this point much longer because it isn't going anywhere and, for the most part, people have responded to it not being a good idea. Fair enough. It is just a discussion after all to see what people think.
> 
> Chalk this up to my naivete for even bringing it up. I know I'm not the only one who has concerns about working on OpenToonz, but I do anyway. Let's leave it at that.
> 
> As for direction...we need direction and more developers willing to work on that direction.
> 
> Not sure what to do to attract more developers. Bounties didn't really do much and the 1 developer who tried to work on a small bounty didn't finish because what we wanted and what he was going to give us wasn't quite the same. Perhaps we need clearer and better requirements?
> 
> Direction/Priority wise: We've done quite a bit to fix the stability and improved some UI/UX issue which is great. I still see a number of users having issues or complaining about Vector levels, mostly with fills and occasional crashes on vector levels. This really needs to be fixed or redone. How can we be taken seriously if something as simple as a fill is problematic?
> 
> From the few issues I've looked into I think a lot of that might have to do with the detection and management of stroke and stroke-intersection data. The problem with isolating the root issue with this is the problems are hard to reliably reproduce making it difficult to see why. In general I think that stroke intersection data should be reviewed and possibly redone.
> 
> I'm still think since we're having some issues because we aren't fully on OpenGL which needs to be implemented properly and tested thoroughly to avoid issues that happened with 1.2 release. I believe we're waiting to see how Qt handles Apple's move to Metal? But how long must we wait? I think we should just move to OpenGL now and then convert over later instead of trying to bypass it completely.
> 
> I'm sure there are many other pain points and improvements that need to be made but without clear leadership and a few more experienced developers, we're not going anywhere.
> 
> Anyway...just my naive thoughts, for what it's worth...

So it turns out that the problem is the lack of developers, but in order to get them, you need to interest them with something ..... it's just that hardly anyone will do something, you need to learn how the blender and other popular free programs developed.

**Comment by Lovewyrm** [comment]
>rigging and SVGs

I, personally, as an individual, do not care one bit about doll animation, or vectors for that matter.
I do, however, appreciate vector capabilities, as in:
Even though I don't use vectors, I recognize their strengths, I merely see them outclassed by their weaknesses (look kind of artificial, awkward to use, etc), but as the blade cuts both ways, that might just be my inexperience due to my reluctance to use them.

That said, I find the raster tools to be really good. They're not fancy (the mypaint ones are, though) but they do the trick for lineart and coloring, at least for me.
I'm definetely in the traditional raster camp, if it exists, but again, I'm a nobody, I just happen to come to really appreicate OT.
(I actually used to be a Krita dev, lol, and I would rather use OT over Krita for animation because I like focused programs... A reason why I tend to prefer Gimp over Krita for retouching, and I *hate* using Gimp lol. )

**Comment by gab3d** [comment]
> So it turns out that the problem is the lack of developers, but in order to get them, you need to interest them with something ..... it's just that hardly anyone will do something, you need to learn how the blender and other popular free programs developed.

@manongjohn 
I understand your points, and currently you are certainly one of the most prominent contributors for OT, so I take them seriously because the point id to retain people like you, and bring more such talent onboard if possible.

I quoted @jrhyme above, because I agree that beyond specific points of what to develop first, and other such technical issues, what should come first is a defined will for developing, both, a path of action, and a slightly more evolved organization structure (I don't think we need anything big to handle these tasks, just the minimum).
And that starts by just simply emulating how other successful projects have been growing and evolving over time, certainly Blender is the bigger/older one (and that's why I've already suggested trying to communicate with them, looking for some advise on successful organizing practices, etc.).

Also, as I see it, we could hardly do any of these type of contacts alone, if we don't have the ability to speak on *behalf of the OpenToonz project* with third parties, and for that I think we should at least have the complacency of whom have been behind the project up to this day (i.e. Dwango).
(In fact I've already talked to someone in the Blender project about this, but won't advance another step in that direction until some concrete intentions are expressed on this regard. 🤔 )

**Comment by konero** [comment]
> @Lovewyrm I do, however, appreciate vector capabilities

Ditto. Here's a couple shots from Dofus, it was animated traditionally and the cleanup is done in Flash. The cleanup is done by using closed vector shapes with fills and they're just keyframed, so there's no need to trace the lines or paint the cels frame by frame. This is something #1878 could probably solve.

![tumblr_o89obk0jjL1see7ilo6_500](https://user-images.githubusercontent.com/19820721/64481089-e1014080-d1cc-11e9-91f5-96b0838e7ad9.gif)

![tumblr_o0r1t6VShd1see7ilo2_500](https://user-images.githubusercontent.com/19820721/64481063-56b8dc80-d1cc-11e9-9bb8-22f868e961ab.gif)

Quite frankly it would be a waste of resources to cleanup something like this frame by frame... heh.

![ezgif-1-2684ffbd81c0](https://user-images.githubusercontent.com/19820721/64481136-d3988600-d1cd-11e9-9297-bb173e2e8917.gif)

So yes vector very important, I didn't mean to say it wasn't, just to be clear. CACANi is doing very well in the vector sector, they will no doubt be the champions soon, but surprisingly they don't have the ability to just tween shapes, yikes.

**Comment by 2dvision** [comment]
@shun-iwasawa  give us two great news:

1. Miyazaki wants to make a new film.
2. Studio Ghibli using OpenToonz for their production.

If this film produce and publish, A lot of people (Artist, student, animation studio, etc) come for test OT and they knows **"OpenToonz"**.

So, you have 1, 2 or 3 years for making OT best performance and quality for all people from different countries and animation techniques.

@konero said:
> So perhaps you may want to allow feature requests again, this way student programmers looking for something to do are not only presented with a long list of bug reports for a software they are yet to dabble in, making reproducing issues much more painful. I know we got Google groups, but coders are unlikely going to visit it.

Is it not possible to create a new repository and called `opentoonz-proposals` and move feature requests, ideas and proposals to this repository And keep reporting bugs here? [Same problem in Godot engine](https://godotengine.org/article/introducing-godot-proposals-repository)

P.S. I can't wait for Miyazaki new film. :))

**Comment by turtletooth** [comment]
@2dvision I saw that Godot had moved their feature requests to a new repo.  I actually think that would work.  @shun-iwasawa Would you be willing to create a OpenToonz proposals repo?

**Comment by jrhyme** [comment]
> I post this comment is to let you know my recent situation and to clarify a policy for reviewers to judge which PR can be merged to OT.
What's up with the color selector? accepted a license or not? there are many questions.

**Comment by jrhyme** [comment]
> Dear OpenToonz community members,
> 
> I post this comment is to let you know my recent situation and to clarify a policy for reviewers to judge which PR can be merged to OT.

What's up with the color selector? accepted a license or not? there are many questions.

**Comment by RodneyBaker** [comment]
> What's up with the color selector? accepted a license?
Currently not accepted.  See the original PR.
In order to use it in OT we have to use OTX, Morevna or other build.

**Comment by jrhyme** [comment]
Brother Shun, will you comment on something?

**Comment by RodneyBaker** [comment]
Shun commented on the Advanced Color Selector in its PR.
See it here:  https://github.com/opentoonz/opentoonz/pull/1370

**Comment by jrhyme** [comment]
> @jrhyme Shun commented on the Advanced Color Selector in its PR.
> See it here: #1370

I'm not interested in the color selector at the moment, are you kidding me? I'm talking about everything here. I turned to a specific person

**Comment by jrhyme** [comment]
> > @jrhyme Shun commented on the Advanced Color Selector in its PR.
> > See it here: #1370
> 
> I'm not interested in the color selector at the moment, I'm talking about everything here. I turned to a specific person. We discuss a lot and the chapter is silent

**Comment by RodneyBaker** [comment]
@jrhyme Perhaps you could restate your specific interest so that Shun doesn't have to guess or wade through the whole topic.  Responding to everything mentioned here will be quite a job.

**Comment by RodneyBaker** [comment]
Rather than start a new topic, this seems to be an appropriate place to ask this.
There are a number of topics labeled as 'Roadmap' which indicates the issue is one of interest moving into the next release.  If something shouldn't be tagged as roadmap then we can untag it and let it slip to a release farther in the future.  **Note that in most cases issues tagged as Roadmap are those that are anticipated for the near future but do not have an active developer working on them.**  As such there will be many active issues/PRs that aren't tagged as 'roadmap' that are actively being pursued.  At present the 'roadmap' tag is designed to put a spotlight on issues/PRs that are not actively being pursued.

Here are the issues currently tagged as 'Roadmap":  [LINK](https://github.com/opentoonz/opentoonz/issues?q=is%3Aissue+is%3Aopen+label%3Aroadmap)

**Comment by jpturcotte** [comment]
I think we could add those to GitHub Milestones.

**Comment by RodneyBaker** [comment]
A few very likely could be dropped (at least as roadmap items).
I suppose a question might be do we expect those currently tagged to be implemented by December/January in time for the v1.4 release?  

Those that we expect to see implemented might retain the roadmap tag with it updated to say 'roadmap (v1.4)'.  Then others could know the issue is still important but needs attention in order to get a bit of life back in for it to be revived.

**Comment by manongjohn** [comment]
Transferring the code review/merging discussion from PR #2659 to here where it is more appropriate...

In all honesty and with all due respect to @RodneyBaker, I too am a bit uncomfortable with code being merged and not being reviewed by someone who knows C++ and is familiar enough with OT's code.

I know my code is not always the greatest. @shun-iwasawa has corrected me numerous times, catching things that were not quite as obvious to me and even offering a way to correct it or at least explain why something might be detrimental.

Unfortunately we do suffer from a lack of active developers with time to support proper code reviews by other developers. So what are our options to keep moving forward until we have more experienced developer-collaborators?  

Personally, I don't feel my C++ is as strong as I would like it to be to review other people's work.  I would not consider myself an expert by any means though I do program for a living, just not in C++ so I am more than capable of doing code reviews. 

Having said that, my knowledge of OT's code base and my knowledge of C++ development has expanded quite a bit since I started so i will try to start reviewing things to help with this issue, but doesn't help with my stuff not getting reviewed by someone else.

**Comment by age3rcm** [comment]
do not know if this helps, but i wish for "better stability" for the future of the program. 
opentoonz already has toonz of features, i fear the issues already present, looking at the issues reported, and if they would ever be fixed

**Comment by RodneyBaker** [comment]
> looking at the issues reported, and if they would ever be fixed

Working on it.

One of the best thing you can do to see those issues resolved is to stay active in them.  When the embers o life die out post a comment on the one's that are most important to you.  In this way you maximize focus on efforts to keep Opentoonz as stable as possible.

No new features isn't likely to happen but the good news is that as new features and enhancements come aboard old code gets a relook and bugs, instabiilty and subobtimization gets addressed in the process.

A good part of development right now is still learning the source code.
The more developers (and users!) know about the code the better.
Then when reports come in they can laser focus on the problem.

**Comment by artisteacher** [comment]
I agree that stability is really important. Stability has improved a lot in the last couple years, but it still has a ways to go. Every semester, the number of times OT crashes becomes something of a joke in my animation classes. Part of that may be due to very old computers, but even on newer devices, there's still a stability issue on macOS. 

Beyond that, a lot of existing features would benefit from further optimization. For example, vector levels have so much potential but in practice are finicky and sometimes are missing standard behaviors/features. Or raster brushes could benefit from some organization...

It would be great to find a way to get more developers on board and excited about OT.

**Comment by artisteacher** [comment]
> having a proper Pen Tool

@gab3d this is a late response, but we almost have a completely functional pen tool with the polyline mode of the geometric tool. The only thing that's missing is the ability to add cusp points by holding down on alt while making a polyline shape. That addition would be amazing - closed shapes made with the geometric tool never have fill issues.

**Comment by manongjohn** [comment]
FYI, I just hid a lot of the comments on this thread as 'Outdated' just to focus the discussion a little more on topic in case someone wants to see our Development direction and not be scared off by a lot of old discussions/reactions.  If I missed any or have hidden something that people think i shouldn't have hidden, I think the original poster of the comment or any Collaborator can unhide.

Moving forward...

I recall some work was done on the GitHub Projects tab to help organize and track issues better.  It would be a good place for Developers to go if they want to focus on fixing certain things. 

The current issue with Projects in github is it doesn't appear to have adequate automation of assigning specific issues to projects.  I found this and think we should consider implementing it:  

     https://github.com/philschatz/project-bot 

It can be configured to automatically add issues to Projects based on labels which would be helpful for us.  Not sure it handles removing from a Project when a label is removed though.

Assuming this works as described, I would suggest starting from scratch and begin with the following Projects based on appropriate labels:
- Crashes and Critical bugs
- Bugs
- Enhancements and Feature Requests
- UI and Translation Issues
- macOS Specific Issues
- Linux Specific Issues

Thoughts? Suggestions?

**Comment by RodneyBaker** [comment]
I'm all for revamping the Projects.
As is they aren't useful or used as far as I can tell.

I'd prefer "Crash / Critical" over adding the word bug in that category.
Rationale:  bug is assumed and everyone believes their bug is critical.
The bugs category is therefore the larger set of bugs that hasn't been designated as critical or that can be demonstrated to produce a crash.
The Crash /Critical category is surely the Project home of the best of the best in Opentoonz development; featuring the most knowledgeable developers who can resolve issues immediately.

In the tagging I have stepped away from the term "Feature Request" in favor of "Feature".  
Rationale:  A feature is either implemented or not.  
Categories inside a Project Kanban then might be "Requested", "Acknowledged", "Assigned", "Implemented"... and some appropriate category for "Still in the Cue"... maybe "Research and Development"(?)

There is a lot of overlap but would be tempted to keep Translation separate from UI specifically because translation has a specific team and focus that allows another team to focus on deeper issues related to the UI (of which there are many).  

I've been looking for ways to place a focus on operating system and am hoping Microsofts move into Linux native kernel might blur the lines even more even to the point where operating system doesn't particularly matter.

In writing this it seems to me that the Projects break down nicely into focus areas for specific teams.  This isn't to say that users/developers can't be in more than one team... we are!  The Projects just help us leverage our interests, skills and focus on specific channels/specialties.

When I first looked at Github Projects it looked limited because very few people were using it or couldld use it.  

I'll look into https://github.com/philschatz/project-bot !
A quick look made me feel like the current projects already have most of this capability.
Perhaps that or a similar project bot is already loaded (?)

P.S.  **If there is anyone currently/actively using the Opentoonz Projects now would be a very good time to identify yourself before anything is changed/removed.**

**Comment by manongjohn** [comment]
There are no project related bots loaded. New issues have to be manually added in the current Projects set up. Looking at options for loading new issues automatically, it doesn't appear to have any granularity or filter in which issues qualify for that project. If you look at the Crash Project there are a lot not listed because someone needs to manually keep up with adding them.  This bot should assist in that, and the remaining Project automation rules can help move to other states within the project, like moving to Completed/Done when an issue is closed.

In my opinion, an issue should be labeled a "Feature **Request**" if it is in fact a request for something new, not an issue with a Feature as a label like that might imply.

We can split the projects as needed. Just don't want 100 of them.  Sometimes less is more.

**Comment by RodneyBaker** [comment]
I'll start a new 'issue' to focus on Opentoonz Projects.

**Comment by hayoko** [comment]
I don't think this will work overall, since this is not donation-ware, getting devs to contribute based on lists and priorities for free on things they are not **personally interested** or vested in will be difficult.

I believe konero explained that in his post (it is a shame it is now hidden), that he held this unique concept for digital animation painting, fleshed it out by discussing it with professionals and offered it on a silver platter but it remains ultimately down to him to develop it, nobody is going to do it on his behalf because it was all of a sudden added to a list, there are no "teams" here.

The problem is concepts like that are just not comprehensible to most people who haven't worked in a real production setting, you would need to have great communication with the original poster. Is that kind of thing going to happen without incentive? Would you really have the patience to work on it?

Here's the thing, that concept was discussed over at TVPaint, one of the guys is from [Wit Studio](https://en.wikipedia.org/wiki/Wit_Studio) who animated Attack on Titan, that is just amazing.

https://forum.tvpaint.com/viewtopic.php?f=21&t=12611

Yet here you guys want to keep having bureaucratic arguments. 😞

**Comment by manongjohn** [comment]
@hayoko 
The Projects use and organization is not meant to force developer to work on something they don't want to. As far as I am concerned, developers can work on whatever they want regardless if on the list or not.   Nothing says you MUST work on these issues before stuff you really want to do.  Contribute what you can however you want. It's all good in the end if it means improving OT.

I'm certainly not getting paid for any of this and doing things because I like OT and want to improve it. If someone TOLD me I had to do XYZ, I'd say FU and only fix stuff for myself.

The Projects, as I see it, are just meant as a way to categorize things for people looking for that kind of stuff to focus on.  I tend to focus on Crashes, Bug fixes and some UI/UX enhancements so having a place to organize that for me is good, instead of wading through the 300+ issues to find what I might be interested in working on.  If people use it, great, if not...that's fine too.

It would be helpful if developers communicated a bit more on what is being worked on. I know at one point I was working on an issue and after a day or so, another developer posted their solution before I finished.  Our solutions were similar, but I wasted my time when I could have been working on something else.

However, I think OT still requires some direction and organization. If a few willing developers wish to work collaboratively on some established goal, great, but I don't think anyone here expects them to only work on theses lists.  Obviously it would go a long way for the community, but definitely not a requirement.

Anyway. I'll just shut up and go back to doing whatever...coding, animating, crying in the corner...

**Comment by RodneyBaker** [comment]
Konero should post that development idea in it's own topic where it can receive full attention.

**Comment by hayoko** [comment]
@manongjohn I was responding to other peoples suggestions of goals, targets etc. which didn't make sense to turn into lists without being able to designate roles to dev teams. I think we have our wires crossed. I wasn't criticizing, I think if you can find a use for Projects then you should do it I just thought if you opened a Project that was titled "Full Vector Curve, Inbetween and Bendy Bones Update March 2020" it would be just a dream.

@RodneyBaker Fair enough, but the topic become much broader so I was I was just jumping in. I deleted the comment as it seems to have been confrontational.

**Comment by hayoko** [comment]
Seems bountysource had some issues awhile back, requires manual fixing to get it going again by a public member of this organization.

You can [read here for instructions](https://github.com/bountysource/core/issues/1096#issuecomment-370379798).

> You'll need to unlock repos that aren't updating.
> 
> Go to https://www.bountysource.com/tools
> 
> You'll find that the project that's not updating to be locked (Let me and @JalenOng know if it isn't!). Just uncheck 'This project is locked' and click save. If all is well, all the issues within that repo should be updated within a few minutes.

I know a lot of devs find their way to respos through places this like, since you guys were asking how to attract attention.

**Comment by jpturcotte** [comment]
@shun-iwasawa It would be great if the Bountysource issue could be resolved.

**Comment by shun-iwasawa** [comment]
@jpturcotte I re-added Bountysource to my Authorized OAuth Apps. Is it enough to resolve the issue?

**Comment by jpturcotte** [comment]
@shun-iwasawa #1807 still has not been updated, so something must not be working correctly.

**Comment by RodneyBaker** [comment]
Closing this as it can still be accessed and discussions can continue in closed topics.

**Comment by RodneyBaker** [comment]
Reopening in light of Opentoonz moving into (very) early stage of preparation for v1.6.
